snuba/migrations/connect.py

```diff
@@ -157,27 +157,17 @@ def _get_all_nodes_for_storage(
     return (local_nodes, distributed_nodes, query_node)
 
 
-def check_for_inactive_replicas(storage_keys: List[StorageKey]) -> None:
+def check_for_inactive_replicas(clusters: List[ClickhouseCluster]) -> None:
     """
     Checks for inactive replicas and raise InactiveClickhouseReplica if any are found.
     """
     checked_nodes = set()
     inactive_replica_info = []
-    for storage_key in storage_keys:
-        try:
-            local_nodes, distributed_nodes, query_node = _get_all_nodes_for_storage(
-                storage_key
-            )
-            storage = get_storage(storage_key)
-            cluster = storage.get_cluster()
-        except UndefinedClickhouseCluster:
-            continue
-
-        for node in (*local_nodes, *distributed_nodes, query_node):
+    for cluster in clusters:
+        for node in cluster.get_local_nodes():
```

Comment:
It seems like previously we were traversing local nodes and query node but now we are only looping over local nodes on the cluster. Any reason to skip query node?

Reply:
The system.replicas table is only available for replicated tables and query nodes use distributed tables which won't have any data in that table.